### PR TITLE
Add has_custom_cutoff_date to logsdb usage.

### PR DESCRIPTION
**Pull request by martijnvg**
Indicates whether `es.mapping.synthetic_source_fallback_to_stored_source.cutoff_date_restricted_override` system property has been configured.

A follow up from #116647

### CODE REVIEW

File: x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceLicenseService.java [hunk 1]
```diff
@@ -129,7 +128,7 @@ private static long getCutoffDate(String cutoffDateAsString) {
             LOGGER.info(
                 "Configuring [{}] to [{}]",
                 CUTOFF_DATE_SYS_PROP_NAME,
-                LocalDateTime.ofInstant(Instant.ofEpochSecond(cutoffDate), ZoneOffset.UTC)
+                LocalDateTime.ofInstant(Instant.ofEpochMilli(cutoffDate), ZoneOffset.UTC)
```
Comment (martijnvg) [inline x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceLicenseService.java:131]: unrelated change. Tiny bug, otherwise date gets printed incorrectly.

File: x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.hamcrest.Matchers;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class LogsdbWithBasicRestIT extends ESRestTestCase {
```
Comment (n1v0lg) [inline x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java:19]: This could just go into the existing `LogsdbWithBasicRestIT` suite or no? Fine with a new suite too, just wondering if it's worth it.

File: x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .systemProperty("es.mapping.synthetic_source_fallback_to_stored_source.cutoff_date_restricted_override", "2027-12-31T23:59")
+        .setting("xpack.license.self_generated.type", "trial")
```
Comment (n1v0lg) [inline x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java:25]:
```suggestion
        .setting("xpack.license.self_generated.type", "trial")
```

I don't think we need this

**Review by n1v0lg — APPROVED** [review]
LGTM - I think we might need to tweak transport versions though. Also an optional suggestion around what we track. 

No need for around round of review though 👍

File: x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java (hunk 2 above)
Comment (martijnvg) [reply x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java:19]: I prefer to do this test in a separate qa module. I don't want other tests by accident rely on the system property set to a specific value.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java [hunk 4]
```diff
@@ -53,13 +62,15 @@ public LogsDBFeatureSetUsage(
...
-        long sizeInBytes
+        long sizeInBytes,
+        boolean hasCustomCutoffDate
     ) {
         super(XPackField.LOGSDB, available, enabled);
         this.indicesCount = indicesCount;
         this.indicesWithSyntheticSource = indicesWithSyntheticSource;
         this.numDocs = numDocs;
         this.sizeInBytes = sizeInBytes;
+        this.hasCustomCutoffDate = hasCustomCutoffDate;
```
[the remark this answers is not in the document]
Comment (martijnvg) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java:73]: My understanding is that we're only interested in learning about clusters that have this system properly set. So I think a boolean should be sufficient here.

File: x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java (hunk 3 above)
Comment (n1v0lg) [reply x-pack/plugin/logsdb/qa/with-custom-cutoff/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbWithBasicRestIT.java:25]: i.e., it can be `"basic"`

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java [hunk 5]
```diff
@@ -74,11 +89,12 @@ protected void innerXContent(XContentBuilder builder, Params params) throws IOEx
         builder.field("indices_with_synthetic_source", indicesWithSyntheticSource);
         builder.field("num_docs", numDocs);
         builder.field("size_in_bytes", sizeInBytes);
+        builder.field("has_custom_cutoff_date", hasCustomCutoffDate);
```
Comment (dnhatn) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java:92]: Can you update the telemetry mappings after this PR?

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Martijn!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java (hunk 5 above)
Comment (martijnvg) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/LogsDBFeatureSetUsage.java:92]: yes, I already have a pr: elastic/telemetry#4260
